pkg/lint/validator.go

```diff
@@ -376,3 +380,103 @@ func (v *Validator) validateDomains(domains ecv1beta1.Domains, allowedDomains []
 
 	return errors
 }
+
+// validateVeleroPlugins validates Velero plugin configurations
+func (v *Validator) validateVeleroPlugins(veleroExt ecv1beta1.VeleroExtensions) []error {
+	var errors []error
+
+	// Check for duplicate plugin names and images
+	seenNames := make(map[string]int)
+	seenImages := make(map[string]int)
+	for i, plugin := range veleroExt.Plugins {
+		// Validate name is not empty (CRD validation should catch this, but double-check)
+		if plugin.Name == "" {
+			errors = append(errors, ValidationError{
+				Field:   fmt.Sprintf("extensions.velero.plugins[%d].name", i),
+				Message: "plugin name is required",
+			})
+		} else {
+			// Check for duplicate names
+			if idx, exists := seenNames[plugin.Name]; exists {
+				errors = append(errors, ValidationError{
+					Field:   fmt.Sprintf("extensions.velero.plugins[%d].name", i),
+					Message: fmt.Sprintf("duplicate plugin name %q (also specified at index %d)", plugin.Name, idx),
+				})
+			} else {
+				seenNames[plugin.Name] = i
+			}
+		}
+
+		// Validate image is not empty (CRD validation should catch this, but double-check)
+		if plugin.Image == "" {
+			errors = append(errors, ValidationError{
+				Field:   fmt.Sprintf("extensions.velero.plugins[%d].image", i),
+				Message: "plugin image is required",
+			})
+			continue
+		}
+
+		// Check for duplicate images
+		if idx, exists := seenImages[plugin.Image]; exists {
+			errors = append(errors, ValidationError{
+				Field:   fmt.Sprintf("extensions.velero.plugins[%d].image", i),
+				Message: fmt.Sprintf("duplicate plugin image %q (also specified at index %d)", plugin.Image, idx),
+			})
+		} else {
+			seenImages[plugin.Image] = i
+		}
+
+		// Validate image format (basic check - should be a valid OCI image reference)
+		if err := v.validateImageFormat(plugin.Image); err != nil {
+			errors = append(errors, ValidationError{
+				Field:   fmt.Sprintf("extensions.velero.plugins[%d].image", i),
+				Message: err.Error(),
+			})
+		}
+	}
+
+	return errors
+}
+
+// validateImageFormat validates that an image string follows a valid OCI image reference format
```

Comment:
is there a standard regex used by tools like docker that we can just use instead of reimplementing this from scratch?

Reply:
There is. I changed this implementation to use oras library for that